crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -103,6 +103,15 @@ pub struct ScopeId<'db> {
 }
 
 impl<'db> ScopeId<'db> {
+    pub(crate) fn is_function_like(self, db: &'db dyn Db) -> bool {
+        matches!(
+            self.node(db),
+            NodeWithScopeKind::ClassTypeParameters(_)
```

Comment:
I would not have expected `ClassTypeParameters` to be in here. A comment might be useful here.

Reply:
This is because the actual values of type parameters (whether they're type parameters for a class or type parameters for a function) are lazily evaluated (so that you can have forward references in them). The way this is implemented internally in Python is just by creating a function that is executed when you try to access the value for the first time. All of which is to say that although the user probably doesn't think of themselves as creating a function when they use type parameters (either for a class for for a function), the scoping rules for type parameter scopes are very analogous to function scopes. Similarly, we'll probably return `true` for this function in the future if `self.node(db)` is a comprehension or generator expression of some kind, once we support those.

Reply:
Yes, these scopes behave like function scopes in terms of name resolution. See e.g. https://github.com/python/cpython/blob/main/Python/symtable.c#L545-L552 in CPython.